chsdi/views/stationboard.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+
+from sqlalchemy import and_
+from pyramid.view import view_config, view_defaults
+from chsdi.models.vector.uvek import OevDepartures
+import datetime
+
+
+@view_defaults(renderer='jsonp', route_name='stationboard')
+class TransportView(object):
+
+    def __init__(self, request):
+        self.request = request
+        if request.matched_route.name == 'stationboard':
+            self.id = request.matchdict['id']
+            self.destination = request.params.get('destination')
+
+    @view_config(request_method='GET')
+    def get_departures(self):
+        current_date = datetime.datetime.now()
+        if self.destination == 'all':
+            query = self.request.db.query(OevDepartures).filter(and_(OevDepartures.stop == self.id, OevDepartures.time > current_date)).order_by(OevDepartures.time).limit(5).count()
+        else:
+            query = self.request.db.query(OevDepartures).filter(and_(OevDepartures.stop == self.id, OevDepartures.destination == self.destination, OevDepartures.time > current_date)).order_by(OevDepartures.time).limit(5).count()
+
+        if query == 0:
+            results = [{
+                'destination': 'nodata'
+            }]
+            return results
+        else:
+            if self.destination == 'all':
+                query2 = self.request.db.query(OevDepartures).filter(and_(OevDepartures.stop == self.id, OevDepartures.time > current_date)).order_by(OevDepartures.time).limit(5)
+            else:
+                query2 = self.request.db.query(OevDepartures).filter(and_(OevDepartures.stop == self.id, OevDepartures.destination == self.destination, OevDepartures.time > current_date)).order_by(OevDepartures.time).limit(5)
+
+            def serialize(time):
+                return time.strftime('%H:%M')
+
+            results2 = [{
```

Comment:
You can use `results` here as well.

Alternatively, you can directly return the array instead of assigning it first to a variable.